ext/openssl/ossl_ts.c

```diff
@@ -1071,7 +1072,10 @@ ossl_ts_token_info_to_der(VALUE self)
 static ASN1_INTEGER *
 ossl_tsfac_serial_cb(struct TS_resp_ctx *ctx, void *data)
 {
-    return (ASN1_INTEGER *)data;
+    ASN1_INTEGER **snptr = (ASN1_INTEGER **)data;
+    ASN1_INTEGER *sn = *snptr;
+    *snptr = NULL;
+    return sn;
```

Comment:
Some context here: This callback is given the serial number by `ossl_tsfac_create_ts`. OpenSSL calls the callback to get the serial number and then eventually frees it for us. If OpenSSL hits an error before it calls the callback though, `ossl_tsfac_create_ts` has no way of knowing that the serial number didn't get freed. So, I'm passing the serial number in as a `ASN1_INTEGER **` so we can avoid leaking or double-freeing in `ossl_tsfac_create_ts`.

Reply:
Maybe you can add this as a comment to the code.